daemon/src/daemon.rs

```diff
@@ -830,24 +837,111 @@ impl Daemon {
 }
 
 async fn spawn_file_watcher(base_dir: PathBuf, document_handle: DocumentActorHandle) {
-    // Beware of the file watching spaghetti triangle monster.
```

Comment:
🍝

Reply:
I think this is _still_ an impressive triangle that might benefit from refactoring. Which I _think_ this comment wanted to say ;-P
<img width="420" alt="image" src="https://github.com/user-attachments/assets/26be8843-f66f-4a5b-8a2c-a4ff1865a4c2">